packages/product-react/src/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,217 @@
+  id: string;
+  ariaLabel: string;
+  label: string;
+}
+
+export interface IProps {
+  tabs: ITab[];
+  defaultActiveTab?: string;
+  onSelect?: (selectedTabId: string) => void;
+  compact?: boolean;
+}
+
+export const Tabs: React.FC<IProps> = props => {
+  const { tabs, onSelect, compact } = props;
+  const [firstTab] = tabs;
+  const defaultTab = props.defaultActiveTab || firstTab.id;
+  const [selectedTabId, setSelectedTabId] = React.useState<string>(defaultTab);
+  const [buttonsFocused, setButtonsFocused] = React.useState(false);
+  const onButtonFocus = () => setButtonsFocused(true);
+  const onButtonBlur = () => setButtonsFocused(false);
+
+  const setActiveTab = (id: string) => () => {
+    setSelectedTabId(id);
+
+    if (onSelect) {
+      onSelect(id);
+    }
+  };
+
+  const { tabRefs, focusedTab } = useFocusManager({
+    tabs,
+    firstFocusedId: defaultTab,
+    buttonsFocused
+  });
+
+  React.useEffect(() => {
+    setActiveTab(focusedTab)();
+  }, [focusedTab]);
+
+  if (tabs.length === 0) {
+    return null;
+  }
+
+  return (
+    <TabContext.Provider value={{ selectedTab: selectedTabId, tabs }}>
+      <div className="ray-tabs__wrapper" role="tablist">
+        {tabs.map(tab => {
+          const tabIsActive = tab.id === selectedTabId;
+
+          return (
+            <div
+              key={tab.id}
+              className={classnames('ray-tabs__button-wrapper', {
+                'ray-tabs__button-wrapper--active': tabIsActive
+              })}
+            >
+              <Button
+                type="tertiary"
+                aria-label={tab.ariaLabel}
+                aria-selected={tabIsActive}
+                onClick={setActiveTab(tab.id)}
+                role="tab"
+                aria-controls={tab.id}
+                ref={tabRefs[tab.id]}
+                tabIndex={tabIsActive ? 0 : -1}
+                data-testid="tab-button"
+                onFocus={onButtonFocus}
+                onBlur={onButtonBlur}
+                compact={compact}
+              >
+                {tab.label}
+              </Button>
+            </div>
+          );
+        })}
+      </div>
+      {props.children}
+    </TabContext.Provider>
+  );
+};
+
+Tabs.displayName = 'Tabs';
+
+export default Tabs;
+
+interface IFocusManager {
+  tabs: ITab[];
+  firstFocusedId: string;
+  buttonsFocused: boolean;
+}
+
+function useFocusManager(props: IFocusManager) {
+  const { tabs, firstFocusedId, buttonsFocused } = props;
+  const [focusedTab, setFocusedTab] = React.useState(firstFocusedId);
+  const [isMounted, setIsMounted] = React.useState(false);
+
+  const tabRefs: ITabRefs = tabs.reduce((acc, { id }) => {
+    return { ...acc, [id]: React.useRef<HTMLButtonElement>(null) };
+  }, {});
+
+  React.useEffect(() => {
+    if (!isMounted) {
+      return setIsMounted(true);
+    }
+
+    const tabRef = tabRefs[focusedTab];
+
+    if (tabRef.current) {
+      tabRef.current.focus();
+    }
+  }, [focusedTab]);
```

Comment:
i think as a rule of thumb with effect hooks any values you reference within should be a dependency or else you'll end up with stale values

Reply:
Can you be more specific? are you talking about isMounted?